### PR TITLE
Swap reverse exchange

### PR DESCRIPTION
**Pull request by luckmer**

### CODE REVIEW

**Review by awojciak — CHANGES_REQUESTED** [review]
too much unnecessary changes

File: src/components/Swap/Swap.tsx [hunk 1]
```diff
@@ -531,19 +540,21 @@ export const Swap: React.FC<ISwap> = ({
...
               <Box className={classes.ableToHover}>
                 <ExchangeRate
-                  tokenFromSymbol={tokens[tokenFromIndex].symbol}
-                  tokenToSymbol={tokens[tokenToIndex].symbol}
+                  onClick={() => {
+                    swapTokens !== null ? setSwapTokens(!swapTokens) : setSwapTokens(true)
+                  }}
+                  tokenFromSymbol={tokens[fromToken].symbol}
+                  tokenToSymbol={tokens[toToken].symbol}
                   amount={swapRate}
```
Comment (awojciak) [inline src/components/Swap/Swap.tsx:556]: ok, I'm gonna once again explain how should look changes here:
1) only one useState - const [rateDirection, setRateDirection] = useState(true)
2) on click rate set rateDirection to its negation, and set it to true, when tokens from and to change
3) amount prop value in ExchangeRate should be swapRate when rateDirection === true and 1/swapRate, when false,
also swap token from and token to symbols
that should be enough changes, rest is unnecessary

**Review by awojciak — CHANGES_REQUESTED** [review]

File: src/components/Swap/Swap.tsx (hunk 1 above)
Comment (luckmer) [reply src/components/Swap/Swap.tsx]: done

File: src/components/Swap/Swap.tsx [hunk 2]
```diff
@@ -531,19 +530,23 @@ export const Swap: React.FC<ISwap> = ({
                 }}
                 exchangeRate={{
                   val: swapRate,
-                  symbol: tokens[tokenToIndex].symbol
+                  symbol: tokens[rateDirection ? tokenFromIndex : tokenToIndex].symbol
```
Comment (awojciak) [inline src/components/Swap/Swap.tsx:533]: not here, only ExchangeRate

File: src/components/Swap/Swap.tsx [hunk 3]
```diff
@@ -109,6 +110,7 @@ export const Swap: React.FC<ISwap> = ({
   const [settings, setSettings] = React.useState<boolean>(false)
   const [detailsOpen, setDetailsOpen] = React.useState<boolean>(false)
   const [inputRef, setInputRef] = React.useState<string>(inputTarget.FROM)
+  const [rateDirection, setRateDirection] = React.useState<boolean | null>(null)
```
Comment (awojciak) [inline src/components/Swap/Swap.tsx:113]: remove null, just use true as initial value

File: src/components/Swap/Swap.tsx [hunk 4]
```diff
@@ -428,6 +429,7 @@ export const Swap: React.FC<ISwap> = ({
               setTokenToIndex(tmp)
               tokens = tokensY
               setTokensY(tokensTmp)
+              setRateDirection(false)
```
Comment (awojciak) [inline src/components/Swap/Swap.tsx:432]: don't set it there

File: src/components/Swap/Swap.tsx [hunk 5]
```diff
@@ -531,19 +530,23 @@ export const Swap: React.FC<ISwap> = ({
...
               />
-
               <Box className={classes.ableToHover}>
                 <ExchangeRate
-                  tokenFromSymbol={tokens[tokenFromIndex].symbol}
-                  tokenToSymbol={tokens[tokenToIndex].symbol}
-                  amount={swapRate}
-                  tokenToDecimals={tokens[tokenToIndex].decimals}
+                  onClick={() => {
+                    rateDirection !== null
```
Comment (awojciak) [inline src/components/Swap/Swap.tsx:542]: setRateDirection(!rateDirection) would be enough

File: src/components/Swap/Swap.tsx (hunk 3 above)
Comment (awojciak) [reply src/components/Swap/Swap.tsx:113]: and now i think [rateReversed, setRateReversed] would be better name

**Review by awojciak — CHANGES_REQUESTED** [review]
also add useEffect changing rateReversed to false each time tokenFromIndex and TokenToIndex change

File: src/components/Swap/Swap.tsx [hunk 6]
```diff
@@ -185,20 +188,25 @@ export const Swap: React.FC<ISwap> = ({
     setInputRef(inputRef === inputTarget.FROM ? inputTarget.TO : inputTarget.FROM)
   }, [swap])
 
+  useEffect(() => {
+    setRateReversed(false)
+  }, [tokenFromIndex, tokenToIndex])
+
+  console.log(rateReversed)
```
Comment (awojciak) [inline src/components/Swap/Swap.tsx:195]: remove this console.log

Comment (luckmer) [reply src/components/Swap/Swap.tsx]: done

**Review by awojciak — APPROVED** [review]

**Pull request merged**